WordPress/gutenberg
# useNavigateRegions: Use closest to determine the next region to navigate to

## What?


This fixes a small bug in region navigation.

## Why?


Better accessibility.

## How?


Replaced:
```
		const selectedIndex = regions.indexOf(
			ref.current.ownerDocument.activeElement
		);
```
This only works if the current element is the region itself so if the user so much as interacts with the content within a region, the user will automatically start at region index 0 or the first region. This is super bad UX. My solution brings us closer to the way it should act with most screen readers but we're still off a bit.
With:
```
		const selectedIndex = regions.indexOf(
			ref.current.ownerDocument.activeElement.closest( '[role="region"]' )
		);
```
By using closest, we'll traverse upwards in the HTML until the wrapping region is found. This gives us the index in the regions array so we can best determine previous/next index.

## Testing Instructions




The following must be tested with a screen reader enabled such as NVDA. I did my testing in Firefox.
1. Open the editor.
2. Press tab to focus the View Posts link.
3. Press ctrl+` until you land on the Open document settings button.
4. Press tab a couple times.
5. Notice how the region navigation is based off where focus is currently placed. This is how screen readers work in-practice, now it is how Gutenberg works.

## Screenshots or screencast
